websockets.md

```diff
@@ -0,0 +1,101 @@
+- [ClientCertificates](https://learn.microsoft.com/en-us/dotnet/api/system.net.websockets.clientwebsocketoptions.clientcertificates?view=net-7.0)
+- [Cookies](https://learn.microsoft.com/en-us/dotnet/api/system.net.websockets.clientwebsocketoptions.cookies?view=net-7.0)
+- [Credentials](https://learn.microsoft.com/en-us/dotnet/api/system.net.websockets.clientwebsocketoptions.credentials?view=net-7.0)
+- [Proxy](https://learn.microsoft.com/en-us/dotnet/api/system.net.websockets.clientwebsocketoptions.proxy?view=net-7.0)
+- [RemoteCertificateValidationCallback](https://learn.microsoft.com/en-us/dotnet/api/system.net.websockets.clientwebsocketoptions.remotecertificatevalidationcallback?view=net-7.0)
+- [UseDefaultCredentials](https://learn.microsoft.com/en-us/dotnet/api/system.net.websockets.clientwebsocketoptions.usedefaultcredentials?view=net-7.0)
+
+This is how you can set up all of these options on the HTTP invoker instance:
+
+```c#
+var handler = new HttpClientHandler();
+handler.CookieContainer = cookies;
+handler.UseCookies = cookies != null;
+handler.ServerCertificateCustomValidationCallback = remoteCertificateValidationCallback;
+handler.Credentials = useDefaultCredentials ?
+    CredentialCache.DefaultCredentials :
+    credentials;
+if (proxy == null)
+{
+    handler.UseProxy = false;
+}
+else
+{
+    handler.Proxy = proxy;
+}
+if (clientCertificates?.Count > 0)
+{
+    handler.ClientCertificates.AddRange(clientCertificates);
+}
+var invoker = new HttpMessageInvoker(handler);
+
+var ws = new ClientWebSocket();
+await ws.ConnectAsync(uri, invoker, cancellationToken);
+```
+
+## WebSockets over HTTP/2
+
+.NET 7 also adds an ability to use WebSocket protocol over HTTP/2, as described in [RFC 8441](https://www.rfc-editor.org/rfc/rfc8441). With that, WebSocket connection would be established over a single stream of HTTP/2 connection. This will allow for a single TCP connection to be shared between two protocols and to be used for several WebSocket connections and HTTP requests at the same time, resulting in more efficient use of the network.
+
+To enable WebSockets over HTTP/2, you can set [ClientWebSocketOptions.HttpVersion](https://learn.microsoft.com/en-us/dotnet/api/system.net.websockets.clientwebsocketoptions.httpversion?view=net-7.0) option to `HttpVersion.Version20`. You can also enable upgrade/downgrade of HTTP version used by setting [ClientWebSocketOptions.HttpVersionPolicy](https://learn.microsoft.com/en-us/dotnet/api/system.net.websockets.clientwebsocketoptions.httpversionpolicy?view=net-7.0) property. These options will behave in the same way `HttpRequestMessage.Version` and `HttpRequestMessage.VersionPolicy` behave.
+
+For example, the following code would probe for HTTP/2 WebSockets, and if a WebSocket connection could not be established, it will fallback to HTTP/1.1:
```

Comment:
```suggestion
For example, the following code would probe for HTTP/2 WebSockets, and if a WebSocket connection cannot be established, it will fallback to HTTP/1.1:
```